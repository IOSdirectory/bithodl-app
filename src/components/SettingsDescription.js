import React, { Component } from 'react';
import { StyleSheet } from 'react-native';
import PropTypes from 'prop-types';
import StyledText from './StyledText';

const styles = StyleSheet.create({
  description: {
    color: '#6D6D72',
    fontSize: 13,
    lineHeight: 17,
    marginTop: -20,
    marginBottom: 30,
    paddingRight: 15,
    paddingLeft: 15
  }
});

export default class SettingsDescription extends Component {
  render() {
    return (
      <StyledText style={styles.description}>
        {this.props.children}
      </StyledText>
    );
  }
}

SettingsDescription.propTypes = {
  children: PropTypes.node
};
