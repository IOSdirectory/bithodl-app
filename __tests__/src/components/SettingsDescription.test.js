import React from 'react';
import { Text } from 'react-native';
import renderer from 'react-test-renderer';
import SettingsDescription from '../../../src/components/SettingsDescription';

describe('SettingsDescription', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <SettingsDescription>
        <Text>698350fe-6f2f-487c-adad-a9f5a7f27b59</Text>
      </SettingsDescription>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
