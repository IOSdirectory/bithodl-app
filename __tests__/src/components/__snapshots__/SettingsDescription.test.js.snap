// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SettingsDescription renders correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  disabled={false}
  ellipsizeMode="tail"
  numberOfLines={undefined}
  style={
    Array [
      Object {
        "color": "#504F63",
        "fontFamily": "Avenir Next",
        "fontWeight": "400",
      },
      Object {
        "color": "#6D6D72",
        "fontSize": 13,
        "lineHeight": 17,
        "marginBottom": 30,
        "marginTop": -20,
        "paddingLeft": 15,
        "paddingRight": 15,
      },
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
  >
    698350fe-6f2f-487c-adad-a9f5a7f27b59
  </Text>
</Text>
`;
